src/main/java/org/openstreetmap/atlas/checks/validation/linear/lines/GeneralizedCoastlineCheck.java

```diff
@@ -0,0 +1,123 @@
+package org.openstreetmap.atlas.checks.validation.linear.lines;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.Location;
+import org.openstreetmap.atlas.geography.Segment;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.LineItem;
+import org.openstreetmap.atlas.geography.atlas.items.Relation;
+import org.openstreetmap.atlas.tags.NaturalTag;
+import org.openstreetmap.atlas.tags.annotations.validation.Validators;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+
+/**
+ * Flags generalized coastlines-- that is, OSM ways with the tag natural=coastline where x% or more
+ * node pairs are y or more meters apart. X and y can be specified in a {@link Configuration}.
+ * Coastlines can be represented in Atlas by {@link LineItem}s and may be members of
+ * {@link Relation}s.
+ *
+ * @author seancoulter
+ */
+public class GeneralizedCoastlineCheck extends BaseCheck<Long>
+{
+    private static final List<String> FALLBACK_INSTRUCTIONS = Collections.singletonList(
+            "This coastline is generalized, as {0}% of node pairs are {1} or more meters apart. To fix, increase the number of nodes along this coastline. (recommended: place a couple more nodes near the dotted portion of the line)");
```

Comment:
Can we be more specific than: "place a couple more nodes"? Another solution could be to indicate that the flagged point is just a midpoint